### ISSUE TITLE
Nested mutable structs are set multiple times to default during deserialize

### ISSUE THREAD

**Issue by jwillemsen** [objectcomputing/OpenDDS#3780]
Extended the DCPS/XTypes test with the following additional IDL

```
@topic
@mutable
struct NestedMutableStruct {
  @id(1) ModifiedNameMutableStruct nested_struct;
};
```

When looking at the generated code the `bool operator>>(Serializer& strm,  ::NestedMutableStruct& stru)` does call `set_default(stru)` which calls `set_default` of `nested_struct), but also in `bool operator>>(Serializer& strm,  ::ModifiedNameMutableStruct& stru)` there is a call to set_default again, so the nested_struct is set to default 2 times, especially with larger nested structures this looks to be a waste of time

**Comment by iguessthislldo** [comment]
I agree this could probably be better. [This is the `set_default`.](https://github.com/objectcomputing/OpenDDS/blob/78332f891b2a8575fddd750439a6ee676e6998c1/dds/idl/marshal_generator.cpp#L2799) This might require removing that, tracking the fields we didn't get, and defaulting those at the end.
